effekt/shared/src/main/scala/effekt/RecursiveDescent.scala

```diff
@@ -380,30 +388,34 @@ class RecursiveDescent(positions: Positions, tokens: Seq[Token], source: Source)
 
   def valDef(): Def =
     nonterminal:
-      ValDef(`val` ~> idDef(), maybeValueTypeAnnotation(), `=` ~> stmt())
+      documented { doc =>
+        ValDef(`val` ~> idDef(), maybeValueTypeAnnotation(), `=` ~> stmt(), doc, span())
+      }
 
   /**
    * In statement position, val-definitions can also be destructing:
    *   i.e. val (l, r) = point(); ...
    */
   def valStmt(): Stmt =
     nonterminal:
+      val doc = maybeDocumentation()
       val startMarker = nonterminal { new {} }
       def simpleLhs() = backtrack {
         `val` ~> idDef() ~ maybeValueTypeAnnotation() <~ `=`
       } map {
         case id ~ tpe =>
           val binding = stmt()
-          val valDef = ValDef(id, tpe, binding).withRangeOf(startMarker, binding)
+          val valDef = ValDef(id, tpe, binding, doc, span()).withRangeOf(startMarker, binding)
```

Comment:
I'm not sure about the spans here, notice the explicit `withRangeOf(startMarker, binding)`. Generally `span()` is always the span of the beginning of the `nonterminal` until the current position. Whenever this is not the span you want, you need to manually create a `Span`.